csrc/moe/moe_align_sum_kernels.cu

```diff
@@ -31,9 +31,17 @@ __global__ void moe_align_block_size_kernel(scalar_t* __restrict__ topk_ids,
   const size_t tokens_per_thread = CEILDIV(numel, blockDim.x);
   const size_t start_idx = threadIdx.x * tokens_per_thread;
 
+  // compute aligned shared mem offset to make sure cumsum is aligned
+  int cnts_byte_offset =
+      ((blockDim.x + 1) * num_experts) * sizeof(token_cnts_t);
+  int aligned_offset =
+      (cnts_byte_offset + sizeof(int32_t) - 1) / sizeof(int32_t);
```

Comment:
How confident are you that https://github.com/vllm-project/vllm/pull/12413 is an alignment issue? My fear is that we are papering over a bug where we're writing off the end of  token_cnts buffer (for example!)